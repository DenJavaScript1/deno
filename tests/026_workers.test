args: --reload run tests/026_workers.ts
output: tests/026_workers.ts.out
