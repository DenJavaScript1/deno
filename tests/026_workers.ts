const jsWorker = new Worker("./tests/subdir/test_worker.js");
const tsWorker = new Worker("./tests/subdir/test_worker.ts");

setTimeout((): void => {
  jsWorker.onmessage = (e): void => {
    console.log("msg from js worker", e);
  };

  tsWorker.onmessage = (e): void => {
    console.log("msg from ts worker", e);
  };

  jsWorker.onerror = (e): void => {
    console.log("error from js worker", e);
  };

  tsWorker.onerror = (e): void => {
    console.log("error from ts worker", e);
  };

  jsWorker.postMessage("hello world");
  tsWorker.postMessage("hello world");
}, 1000);

setTimeout((): void => {
  jsWorker.postMessage("hello js after timeout");
}, 2250);

setTimeout((): void => {
  tsWorker.postMessage("hello ts after timeout");
}, 2500);
