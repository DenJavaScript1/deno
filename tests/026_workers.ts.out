msg from js worker js worker init
msg from ts worker ts worker init
js worker received message: hello world
ReferenceError: DoesNotExist is not defined
    at onmessage ([WILDCARD]/tests/subdir/test_worker.js:10:5)
    at workerMain (gen/cli/bundle/main.js:[WILDCARD])
ts worker received message: hello world
Error: error from ts worker
    at onmessage ([WILDCARD]/tests/subdir/test_worker.ts:7:15)
    at workerMain (gen/cli/bundle/main.js:[WILDCARD])
msg from js worker js worker init
msg from ts worker ts worker init
error from js worker { message: "DoesNotExist is not defined" }
error from ts worker { message: "error from ts worker" }
js worker received message: hello js after timeout
end js worker
ts worker received message: hello ts after timeout
end ts worker